chess2.cpp

```diff
@@ -0,0 +1,882 @@
+        }
+        if (row == 1 && board[row + 1][col] == 0 && board[row + 2][col] == 0){
+            pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 2, col));
+        }
+        if (board[row + 1][col] == 0)
+            pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 1, col));
+        if (col != 0 && board[row + 1][col - 1] > 0)
+            pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 1, col - 1));
+        if (col != 7 && board[row + 1][col + 1] > 0)
+            pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 1, col + 1));
+    }
+
+    //returnpseudomoves;
+}
+
+void Chess2::getkingmoves(bool turn, int row, int col) {
+
+    if (!turn) {
+        if (row != 7 && board[row + 1][col] >= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 1, col));
+        if (row != 0 && board[row - 1][col] >= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row - 1, col));
+        if (col != 7 && board[row][col + 1] >= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row, col + 1));
+        if (col != 0 && board[row][col - 1] >= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row, col - 1));
+        if (row != 7 && col != 7 && board[row + 1][col + 1] >= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 1, col + 1));
+        if (row != 7 && col != 0 && board[row + 1][col - 1] >= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 1, col - 1));
+        if (row != 0 && col != 7 && board[row - 1][col + 1] >= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row - 1, col + 1));
+        if (row != 0 && col != 0 && board[row - 1][col - 1] >= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row - 1, col - 1));
+    }
+    else if (turn) {
+        if (row != 7 && board[row + 1][col] <= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 1, col));
+        if (row != 0 && board[row - 1][col] <= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row - 1, col));
+        if (col != 7 && board[row][col + 1] <= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row, col + 1));
+        if (col != 0 && board[row][col - 1] <= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row, col - 1));
+        if (row != 7 && col != 7 && board[row + 1][col + 1] <= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 1, col + 1));
+        if (row != 7 && col != 0 && board[row + 1][col - 1] <= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row + 1, col - 1));
+        if (row != 0 && col != 7 && board[row - 1][col + 1] <= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row - 1, col + 1));
+        if (row != 0 && col != 0 && board[row - 1][col - 1] <= 0) pseudomoves.insert(pseudomoves.begin(),push(row, col, row - 1, col - 1));
+    }
+    //returnpseudomoves;
+}
+
+
+int Chess2::evaluation() {
+    const short int pawn = 95,bishop = 330,knight = 320,rook = 500,queen = 900,king = 2000;
```

Comment:
constant variable should be uppercase.
PAWN, BISHOP, etc

Reply:
And can be kept a separate file and imported here
800 lines of code can be broken down into logical file and imported in the main file which has main function